common/utils.go

```diff
@@ -55,3 +59,18 @@ func ValidateDisplayName(displayName *string) error {
 
 	return nil
 }
+
+// implementation referenced from https://github.com/embarklabs/embark/blob/master/packages/plugins/ens/src/index.js
+func IsENSName(displayName string) bool {
+	if len(displayName) == 0 {
+		return false
+	}
+
+	for _, ensExt := range ENS_ALLOWLIST {
```

Comment:
This is not entirely true. We block above `_eth` and `-eth` because they look like ENS names, but they are not. Only `.eth` is a real ENS name.

So here you should only check for `.eth`